src/App.tsx

```diff
@@ -1,14 +1,65 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { SortType } from './types/SortType';
+import { User } from './types/User';
 
 export const App: React.FC = () => {
+  const [loading, setLoading] = useState(false);
+  const [showModal, setShowModal] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [currentTodo, setCurrentTodo] = useState<Todo | null>(null);
+  const [user, setUser] = useState<User | null>(null);
+  const [sortType, setSortType] = useState(SortType.ALL);
+
+  useEffect(() => {
+    setLoading(true);
+
+    getTodos()
+      .then(setTodos)
+      .finally(() => setLoading(false));
+  }, []);
+
+  const getPreparedTodos = useCallback((
+    currentTodos: Todo[],
+  ) => {
+    let sortedTodos: Todo[] = [];
+
+    switch (sortType) {
+      case SortType.ALL:
+        sortedTodos = [...currentTodos];
+        break;
+
+      case SortType.ACTIVE:
+        sortedTodos = currentTodos.filter(todo => !todo.completed);
+        break;
+
+      case SortType.COMPLETED:
+        sortedTodos = currentTodos.filter(todo => todo.completed);
+        break;
+
+      default:
+        throw new Error('Wrong sort type');
+    }
+
+    return sortedTodos.filter(todo => todo.title.toLowerCase().includes(inputValue));
+  }, [inputValue, sortType]);
+
+  const preparedTodos = getPreparedTodos(todos);
```

Comment:
try to use `useMemo` hook
```suggestion
  const preparedTodos = useMemo(...)
```